--- a/plugin_python.c	2018-04-06 19:37:26.000000000 +0200
+++ b/plugin_python.c	2018-04-06 19:39:01.000000000 +0200
@@ -55,7 +55,7 @@
     const char *rv = NULL;
     int i;
 
-    pName = PyString_FromString(module);
+    pName = PyUnicode_FromString(module);
     /* Error checking of pName left out */
 
     pModule = PyImport_Import(pName);
@@ -71,7 +71,7 @@
 	if (pFunc && PyCallable_Check(pFunc)) {
 	    pArgs = PyTuple_New(argc);
 	    for (i = 0; i < argc; ++i) {
-		pValue = PyString_FromString(argv[i]);
+                pValue = PyUnicode_FromString(argv[i]);
 		if (!pValue) {
 		    Py_DECREF(pArgs);
 		    Py_DECREF(pModule);
@@ -85,7 +85,7 @@
 	    pValue = PyObject_CallObject(pFunc, pArgs);
 	    Py_DECREF(pArgs);
 	    if (pValue != NULL) {
-		rv = PyString_AsString(pValue);
+		rv = PyUnicode_AsUTF8(pValue);
 		SetResult(&result, R_STRING, rv);
 		Py_DECREF(pValue);
 		/* rv is now a 'dangling reference' */
@@ -128,7 +128,7 @@
 	Py_Initialize();
 	python_cleanup_responsibility = 1;
     }
-    AddFunction("python::exec", 3, my_exec);
+    AddFunction("python3::exec", 3, my_exec);
     return 0;
 }
 
