diff --git a/work/Core-3.3.0/src/bindings/CMakeLists.txt b/work/Core-3.3.0/src/bindings/CMakeLists.txt
index f7b723a..3cfacfd 100644
--- a/src/bindings/CMakeLists.txt
+++ b/src/bindings/CMakeLists.txt
@@ -150,7 +150,7 @@ if (WITH_YAF_RUBY_BINDINGS)
 
 	endif(NOT WIN32)
 
-	include_directories(${YAF_INCLUDE_DIRS} ${RUBY_INCLUDE_DIR})
+	include_directories(${YAF_INCLUDE_DIRS} ${RUBY_INCLUDE_DIR} ${RUBY_ARCHHDR_DIR})
 	include(${SWIG_USE_FILE})
 
 	set(CMAKE_SWIG_FLAGS -c++ -w-801)
